File: behavysis/pydantic_models/bouts.py
"""
_summary_
"""

from typing import Dict

from behavysis.utils.pydantic_base_model import PydanticBaseModel


class Bout(PydanticBaseModel):
    start: int
    stop: int
    dur: int
    behav: str
    actual: int
    user_defined: Dict[str, int]


class BoutStruct(PydanticBaseModel):
    behav: str
    user_defined: list[Bout]


class Bouts(PydanticBaseModel):
    start: int
    stop: int
    bouts: list[Bout]
    bouts_struct: list[BoutStruct]

File: behavysis/viewer-web/src/renderer/models/Bouts.ts
import { Bout, BoutsData, BoutStruct } from './Bout';

export class BoutsModel {
  data: BoutsData;

  constructor(data?: BoutsData) {
    this.data = data || {
      start: 0,
      stop: 0,
      bouts: [],
      bouts_struct: [],
    };
  }

  get bouts() {
    return this.data.bouts;
  }

  set bouts(val: Bout[]) {
    this.data.bouts = val;
  }

  get bouts_struct() {
    return this.data.bouts_struct;
  }

  updateBout(index: number, updates: Partial<Bout>) {
    if (index >= 0 && index < this.data.bouts.length) {
      this.data.bouts[index] = { ...this.data.bouts[index], ...updates };
    }
  }
}
